layout/layout.h

```diff
@@ -746,6 +747,18 @@ M0_INTERNAL struct m0_layout *m0_layout_find(struct m0_layout_domain *dom,
 M0_INTERNAL int64_t m0_layout_find_by_buffsize(struct m0_layout_domain *dom,
 					       struct m0_fid *pver,
 					       size_t buffsize);
+
+/**
+ * Find optimal layout id by pool_id and supposed object size. It's
+ * another variant of m0_layout_find_by_buffsize() which might be easier
+ * to use. The returned layout id can be passed straight away to the
+ * m0_obj_init().
+ *
+ * Returns -EINVAL if cannot find the right layout.
+ */
+M0_INTERNAL int64_t m0_layout_find_by_objsz(struct m0_client *cli,
+					    struct m0_fid *pool, size_t sz);
```

Comment:
m0_layout_find_by_objsz() and m0_layout_find_by_buffsize() are similar.
Why do they have different parameters?
I mean, the m0_layout_find_by_buffsize() has layout domain and pool version, but the m0_layout_find_by_objsz() has m0_client and pool id.
So, they are used in different environment? Or for different purposes?

Reply:
As I have written above in the description - it might be easier and more intuitive to use m0_layout_find_by_objsz() than m0_layout_find_by_buffsize(). Compare the parameters: layout domain and pool version are very internal to Motr. It's not quite clear where to get the values for these parameters, one have to dig deeply into Motr internals and check the examples. In case of m0_layout_find_by_objsz() all parameters are clear: 1) m0_client instance; 2) pool id; 3) object size (to the best user knowledge). That's all. Easy!

Reply:
Yes. That's what I am thinking.